pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,501 @@
+from typing import Dict, List, Optional, Set
+
+from LoadSWC import SWCGraph, SWCNode
+
+
+class NeuroMLWriter:
+    def __init__(self, swc_graph: "SWCGraph"):
+        """
+        Initialize the NeuroMLWriter.
+
+        :param swc_graph: The SWC graph containing the neuronal morphology.
+        """
+        print("Initializing NeuroMLWriter")
+        self.swc_graph: "SWCGraph" = swc_graph
+        self.points: List[SWCNode] = swc_graph.nodes
+        self.section_types: List[str] = SWCNode.TYPE_NAMES
+        self.morphology_origin: str = swc_graph.metadata.get(
+            "ORIGINAL_SOURCE", "Unknown"
+        )
+        self.seg_per_typ: List[int] = [0] * len(self.points)
+        self.segment_content: List[str] = []
+        self.group_content: List[str] = []
+        self.cable_ids_vs_indices: Dict[int, int] = {}
+        self.point_indices_vs_seg_ids: Dict[int, int] = {}
+        self.segment_groups: Dict[str, List[str]] = {}
+        self.next_segment_id: int = 0
+        self.next_cable_id: int = 0
+        self.cable_prefix_v2: str = "Cable_"
+        self.verbose: bool = True
+        self.processed_nodes: Set[int] = set()
+        print(f"NeuroMLWriter initialized with {len(self.points)} points")
+
+    def nml_string(self, version: str) -> str:
+        """
+        Generate the NeuroML string representation.
+
+        :param version: The NeuroML version to use.
+        :return: The NeuroML string representation.
+        """
+        print("Starting NeuroML generation")
+        if (
+            len(self.points) < 2
+            or len(self.section_types) < 2
+            or self.section_types[1].lower() != "soma"
+        ):
+            print("Error: null data or section types in nmlWrite")
+            return ""
+
+        cell_name = self.get_cell_name()
+        start_point = self.find_start_point()
+
+        print(f"Cell name: {cell_name}")
+        print(f"Start point: {start_point}")
```

Comment:
if you're using these for debugging, use `logging` so that you can use `logger.debug` and so on. Only stuff that is definitely needed to be printed to the user should use `print`.